src/Blockcore.Indexer.Core/Storage/Mongo/MongoStorageOperations.cs

```diff
@@ -202,9 +189,29 @@ public SyncBlockInfo PushStorageBatch(StorageBatch storageBatch)
          }
 
          var unspentOutputTableTask = utxos.Any()
-            ? data.UnspentOutputTable.InsertManyAsync(utxos)
+            ? data.UnspentOutputTable.InsertManyAsync(utxos, new InsertManyOptions { IsOrdered = false })
             : Task.CompletedTask;
 
+         var inputTableTask = Task.CompletedTask;
+         if (storageBatch.InputTable.Any())
+         {
+            var utxosLookups = FetchUtxos(
+               storageBatch.InputTable
+                  .Where(_ => _.Address == null)
+                  .Select(_ => _.Outpoint));
+
+            foreach (InputTable input in storageBatch.InputTable)
+            {
+               if (input.Address != null) continue;
+
+               string key = input.Outpoint.ToString();
```

Comment:
didn't we way we will use outpoint hash function for comparisons over string? (future change?)